frontend/src/component/admin/network/NetworkTrafficUsage/PeriodSelector.tsx

```diff
@@ -0,0 +1,230 @@
+import { styled } from '@mui/material';
+import { type FC, useState } from 'react';
+
+export type Period = {
+    key: string;
+    dayCount: number;
+    label: string;
+    year: number;
+    month: number;
+    selectable: boolean;
+    shortLabel: string;
+};
+
+export const toSelectablePeriod = (
+    date: Date,
+    label?: string,
+    selectable = true,
+): Period => {
+    const year = date.getFullYear();
+    const month = date.getMonth();
+    const period = `${year}-${(month + 1).toString().padStart(2, '0')}`;
+    const dayCount = new Date(year, month + 1, 0).getDate();
+    return {
+        key: period,
+        year,
+        month,
+        dayCount,
+        shortLabel: date.toLocaleString('en-US', {
+            month: 'short',
+        }),
+        label:
+            label ||
+            date.toLocaleString('en-US', { month: 'long', year: 'numeric' }),
+        selectable,
+    };
+};
+
+const currentDate = new Date(Date.now());
+const currentPeriod = toSelectablePeriod(currentDate, 'Current month');
+
+const getSelectablePeriods = (): Period[] => {
+    const selectablePeriods = [currentPeriod];
+    for (
+        let subtractMonthCount = 1;
+        subtractMonthCount < 12;
+        subtractMonthCount++
+    ) {
+        // JavaScript wraps around the year, so we don't need to handle that.
+        const date = new Date(
+            currentDate.getFullYear(),
+            currentDate.getMonth() - subtractMonthCount,
+            1,
+        );
+        selectablePeriods.push(
+            toSelectablePeriod(date, undefined, date > new Date('2024-03-31')),
+        );
+    }
+    return selectablePeriods;
+};
```

Comment:
These functions are mostly copied from `frontend/src/hooks/useTrafficData.ts`, but slightly modified. We'll probably want to reconsider this as the API is built out. I don't want to touch them more than I need to right now.